Tests/Binary/Loader/ChainLoaderTest.php

```diff
@@ -0,0 +1,138 @@
+{
+    public function testConstruction()
+    {
+        $this->getChainLoader();
+    }
+
+    public function testImplementsLoaderInterface()
+    {
+        $this->assertInstanceOf('\Liip\ImagineBundle\Binary\Loader\LoaderInterface', $this->getChainLoader());
+    }
+
+    /**
+     * @return array[]
+     */
+    public static function provideLoadCases()
+    {
+        $file = pathinfo(__FILE__, PATHINFO_BASENAME);
+
+        return array(
+            array(
+                __DIR__,
+                $file,
+            ),
+            array(
+                __DIR__.'/',
+                $file,
+            ),
+            array(
+                __DIR__, '/'.
+                $file,
+            ),
+            array(
+                __DIR__.'/../../Binary/Loader',
+                '/'.$file,
+            ),
+            array(
+                realpath(__DIR__.'/..'),
+                'Loader/'.$file,
+            ),
+            array(
+                __DIR__.'/../',
+                '/Loader/../../Binary/Loader/'.$file,
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider provideLoadCases
+     *
+     * @param string $root
+     * @param string $path
+     */
+    public function testLoad($root, $path)
+    {
+        $this->assertValidLoaderFindReturn($this->getChainLoader(array($root))->find($path));
+    }
+
+    /**
+     * @return array[]
+     */
+    public function provideInvalidPathsData()
+    {
+        return array(
+            array('../Loader/../../Binary/Loader/../../../Resources/config/routing.yaml'),
+            array('../../Binary/'),
+        );
+    }
+
+    /**
+     * @dataProvider provideInvalidPathsData
+     *
+     * @expectedException \Liip\ImagineBundle\Exception\Binary\Loader\NotLoadableException
+     * @expectedExceptionMessage Source image not resolvable
+     */
+    public function testThrowsIfFileDoesNotExist($path)
+    {
+        $this->getChainLoader()->find($path);
+    }
+
+    /**
+     * @return FileSystemLocator
+     */
+    private function getFileSystemLocator()
+    {
+        return new FileSystemLocator();
+    }
+
+    /**
+     * @param string[] $paths
+     *
+     * @return ChainLoader
+     */
+    private function getChainLoader(array $paths = array())
+    {
+        return new ChainLoader(array(
+            'foo' => new FileSystemLoader(
+                MimeTypeGuesser::getInstance(),
+                ExtensionGuesser::getInstance(),
+                $paths ?: array(__DIR__),
+                $this->getFileSystemLocator()
+            )
+        ));
+    }
+
+    /**
+     * @param FileBinary|mixed $return
+     * @param string|null      $message
+     */
+    private function assertValidLoaderFindReturn($return, $message = null)
+    {
+        $this->assertInstanceOf('\Liip\ImagineBundle\Model\FileBinary', $return, $message);
+        $this->assertStringStartsWith('text/', $return->getMimeType(), $message);
```

Comment:
Not a change, just curiosity--Is this assertion purely for the case where something is returned?